### PR TITLE
feat: Force delete support

### LINKED ISSUE #225: feat: Add support for force delete
Stacks that have not been created successfully (terraform code issues), cannot be deleted through CLI.

### PR DESCRIPTION
**Pull request by alexlokshin-czi**
Fixes #225 and [CORE-1]

[CORE-1]: https://czi-eng.atlassian.net/browse/CORE-1?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: pkg/backend/aws/ecs.go [hunk 1]
```diff
@@ -358,23 +359,33 @@ func (ab *Backend) getTaskId(taskArn string) (string, error) {
...
 		logGroup, ok := containerDefinition.LogConfiguration.Options[AwsLogsGroup]
 		if !ok {
 			continue
 		}
-		logStreamName := taskId
+		logStreamName := *container.RuntimeId
+
 		logPrefix, ok := containerDefinition.LogConfiguration.Options[AwsLogsStreamPrefix]
 		if ok {
 			logStreamName = path.Join(logPrefix, *containerDefinition.Name, taskId)
```
Comment (edulop91) [inline pkg/backend/aws/ecs.go:386]: why do we still use `taskId` here?

Comment (edulop91) [inline pkg/backend/aws/ecs.go:386]: why do we still use `taskId` here?

File: pkg/backend/aws/logs.go [hunk 2]
```diff
@@ -13,6 +14,8 @@ func (b *Backend) GetLogs(
 	input *cloudwatchlogs.GetLogEventsInput,
 	f GetLogsFunc,
 ) error {
+	log.Infof("cloudwatch log group: '%s', log stream: '%s'", *input.LogGroupName, *input.LogStreamName)
```
Comment (edulop91) [inline pkg/backend/aws/logs.go:17]:
```suggestion
	log.Debugf("cloudwatch log group: '%s', log stream: '%s'", *input.LogGroupName, *input.LogStreamName)
```
```suggestion
	log.Infof("cloudwatch log group: '%s', log stream: '%s'", *input.LogGroupName, *input.LogStreamName)
```

File: pkg/workspace_repo/workspace.go [hunk 3]
```diff
@@ -91,6 +91,27 @@ func (s *TFEWorkspace) Run(isDestroy bool) error {
...
+	options := tfe.StateVersionListOptions{
+		ListOptions: tfe.ListOptions{
+			PageNumber: 0,
+			PageSize:   10,
+		},
+		Organization: s.workspace.Organization.Name,
+		Workspace:    s.WorkspaceName(),
+	}
+	list, err := s.tfc.StateVersions.List(ctx, &options)
+	if err != nil {
```
Comment (edulop91) [inline pkg/workspace_repo/workspace.go:104]: double err check here not needed

Comment (edulop91) [inline pkg/workspace_repo/workspace.go:104]: double err check here not needed

File: pkg/backend/aws/ecs.go (hunk 1 above)
Comment (alexlokshin-czi) [reply pkg/backend/aws/ecs.go:386]: EC2 profile tasks rely on taskId, we can completely deprecate it

Comment (alexlokshin-czi) [reply pkg/backend/aws/ecs.go:386]: Streams are named differently for EC2 and Fargate tasks

File: cmd/delete.go [hunk 4]
```diff
@@ -70,45 +75,69 @@ func runDelete(cmd *cobra.Command, args []string) error {
...
+		log.Info("No state found for stack, workspace will be removed")
+		return removeWorkspace(ctx, stackService, stackName)
+	}
+
 	// Destroy the stack
 	destroySuccess := true
 	if err = stack.Destroy(ctx); err != nil {
 		// log error and set a flag, but do not return
-		log.Infof("Failed to destroy stack %s", err)
+		log.Infof("Failed to destroy stack: '%s'", err)
```
Comment (edulop91) [inline cmd/delete.go:105]:
```suggestion
		log.Errorf("Failed to destroy stack: '%s'", err)
```

**Review by edulop91 — APPROVED** [review]

**Pull request merged**